### PR TITLE
chore: bump actions/checkout to v4

### PR DESCRIPTION
**Pull request by louib**

### CODE REVIEW

**Review by droidmonkey — APPROVED** [review]

**Comment by sseemayer** [comment]
@louib the master branch is set up to require pull requests, passing status checks, up-to-date branches and a linear history while disallowing force pushes and deletions. Do you mean to also require approvals? I am setting it to require at least one approval before merging now.

**Comment by louib** [comment]
> Do you mean to also require approvals? I am setting it to require at least one approval before merging now.

My bad, that's what I meant indeed. Thank you :pray:

**Pull request merged**
